flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,185 @@
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'enable_encrypted_communication': {'type': 'boolean', 'default': False},
+            'certificates': {
+                'type': 'object',
+                'properties': {
+                    'broker_ca_cert': {'type': 'string'},
+                    'client_cert': {'type': 'string'},
+                    'client_key': {'type': 'string'},
+                    'validate_broker_cert': {'type': 'boolean', 'default': True},
+                    'tls_version': {'type': 'string', 'enum': ['tlsv1.2', 'tlsv1.1', 'tlsv1']},
+                },
+                'additionalProperties': False,
+            },
+            'qos': {'type': 'integer', 'minimum': 0, 'maximum': 2, 'default': 0},
+            "retain": {"type": "boolean", 'default': False},
+        },
+        'additionalProperties': False,
+        'required': ['broker_address','topic'],
+    }
+       
+    def notify(self, title, message, config):
+        """
+        Publish to an MQTT topic
+        """
+        try:
+            import paho.mqtt.client as mqtt
+
+        except ImportError as e:
+            logger.verbose('Error importing paho.mqtt.client: {}', e)
+            raise DependencyError(
+                plugin_name, 'paho.mqtt.client', 'paho-mqtt python module is required for MQTT notify plugin. ImportError: %s' % e
+            )
+
+        def on_log_cb(client, userdata, level, buff):
+            logger.verbose(str(buff))
+
+        def on_publish_cb(client, userdata, mid):
+            logger.verbose('MQTT on_publish callback -  message was successfully published to broker as messageID={}',str(mid))
+            client.disconnect()
+
+        def on_disconnect_cb(client, userdata, rc):
+            logger.verbose('MQTT on_disconnect callback - disconnected with result code {} [{}]',str(rc),conn_rc_description_map.get(rc),'Unknown')
+            client.loop_stop()
+            
+        config['title'] = title
+        config['message'] = message
+        config['payload'] = '{} - {}'.format(config.get('title'),config.get('message'))
+
+        conn_rc_description_map = { 0: 	'Connection Accepted',
+                            1: 	'Connection Refused, unacceptable protocol version - The Server does not support the level of the MQTT protocol requested by the Client',
+                            2: 	'Connection Refused, identifier rejected - The Client identifier is correct UTF-8 but not allowed by the Server',
+                            3: 	'Connection Refused, Server unavailable - The Network Connection has been made but the MQTT service is unavailable',
+                            4: 	'Connection Refused, bad user name or password - The data in the user name or password is malformed',
+                            5: 	'Connection Refused, not authorized - The Client is not authorized to connect' }
+
+        MQTT_proto_map = { 'MQTTv31' : mqtt.MQTTv311,
```

Comment:
since this is the same values as the `broker_protocol`, you could move this outside the class and use its keys as the enum itself:
```python
'broker_protocol': {'type': 'string', 'default': 'MQTTv311', 'enum': list(MQTT_proto_map)},
```

Reply:
This is also nice because it keeps it from cluttering up the logic.

Reply:
Addressed with commit 80d1b42 to cleanup and simplify code